R/check.R

```diff
@@ -1,91 +1,80 @@
-      name = name, value = value, class = "invalid_argument"
-    )
-  }
+  null.ok <- !required
+  assert_string(value, null.ok = null.ok, .var.name = name)
 }
 
-check_int_param <- function(name, value, required = TRUE) {
-  # numeric or numeric[]
-  if ((required &&
-    !(!is.null(value) &&
-      is.numeric(value))) ||
-    (!required && !(is.null(value) || is.numeric(value)))) {
-    rlang::abort(paste0("argument ", name, " is not a number"),
-      name = name, value = value, class = "invalid_argument"
-    )
-  }
+#' Allows string vectors of length > 1 and lists of string vectors of length 1
+#' @importFrom checkmate assert_character
+check_character_param <- function(name, value, required = TRUE) {
+  null.ok <- !required
+  assert_character(value, null.ok = null.ok, any.missing = FALSE, .var.name = name)
 }
 
-is_epirange_like <- function(value) {
-  # character or numeric or EpiRange or list(from=..., to=...)
-  is.character(value) ||
-    is.numeric(value) ||
-    inherits(value, "EpiRange") ||
-    (is.list(value) &&
-      "from" %in% names(value) && "to" %in% names(value))
+#' Allows a single numeric value
+#' @importFrom checkmate assert_integerish
+check_scalar_integerish_param <- function(name, value, required = TRUE) {
+  null.ok <- !required
+  assert_integerish(value, len = 1, null.ok = null.ok, .var.name = name)
 }
 
-check_single_epirange_param <- function(name, value, required = TRUE) {
-  if ((required &&
-    !(!is.null(value) &&
-      is_epirange_like(value))) ||
-    (!required && !(is.null(value) || is_epirange_like(value)))) {
-    rlang::abort(paste0("argument ", name, " is not a epirange"),
-      name = name, value = value, class = "invalid_argument"
-    )
-  }
+#' Allows a vector of numeric values or a list of numeric values
+#' @importFrom checkmate assert_integerish
+check_integerish_param <- function(name, value, required = TRUE) {
+  null.ok <- !required
+  assert_integerish(value, null.ok = null.ok, any.missing = FALSE, .var.name = name)
 }
 
-check_epirange_param <- function(name, value, required = TRUE) {
-  if (is.null(value)) {
-    if (required) {
-      rlang::abort(paste0("argument ", name, " is not a epirange"),
-        name = name, value = value, class = "invalid_argument"
-      )
-    }
-    return()
-  }
-  if (is_epirange_like(value)) {
-    return()
-  }
-  if (!is.list || !all(sapply(sapply(value, is_epirange_like)))) {
-    rlang::abort(paste0("argument ", name, " is not a epirange"),
-      name = name, value = value, class = "invalid_argument"
-    )
-  }
+#' Allows a scalar date_like param: a single date, character, or integerish
+#' @importFrom checkmate assert check_date check_character check_integerish
+check_scalar_date_param <- function(name, value, required = TRUE) {
+  null.ok <- !required
+  assert(
+    # check_date(value, null.ok = null.ok, len = 1),
+    check_character(value, null.ok = null.ok, len = 1),
+    check_integerish(value, null.ok = null.ok, len = 1),
```

Comment:
todo: Date support should allow `Date` objects here

Reply:
todo: test endpoint/request behavior with hyphenated date strings and Dates.  I expect this to break.  Fix for Dates is probably `nonhyphenated_datestring <- format(date, "%Y%m%d")`.  And hyphenated strings could either convert to Date and do the same, or substring to get the nonhyphenated form (the former would be stricter client-side, as conversion to Date would do some extra validation).